src/bin/common/solving_loop.ml

```diff
@@ -489,7 +489,17 @@ let main () =
       | {contents = `Get_model; _ } ->
         if Options.get_interpretation () then
           match State.get partial_model_key st with
-          | Some partial_model -> SAT.get_model partial_model; st
+          | Some partial_model ->
+            begin
+              match SAT.get_model partial_model with
+              | Some (lazy model) ->
+                Models.output_concrete_model
+                  (Options.Output.get_fmt_regular ()) model;
+                st
+              | _ ->
+                (* TODO: is it reachable? *)
+                st
+            end
```

Comment:
It's done.

Reply:
Besides, the `partial_model_key` is now an option on a closure of the model. But we will need again to access to the last consistent environment in order to implement `get-info` in `Solving_loop` (Steven is working on it). A better design consists in modifying the signature of `get_model` in the module `Sat_sig`.  The function should return a `Models.t option` instead of `Models.t Lazy.t option`. If you call `get_model`, you want it and the value has to be force in `get_model` directly and keep the last consistent environment in `partial_model_key`. We should rename this key by the way as I will use it for different thing. Maybe `consistent_env_key`?

Reply:
As explained offline, this modification will cause a conflict with Steven's PR. I reverted it.